src/globus_sdk/services/search/client.py

```diff
@@ -359,13 +356,16 @@ def post_search(
                     :ref: search/reference/post_query/
         """
         log.debug(f"SearchClient.post_search({index_id}, ...)")
-        add_kwargs = {}
-        if offset is not None:
-            add_kwargs["offset"] = offset
-        if limit is not None:
-            add_kwargs["limit"] = limit
-        if add_kwargs:
-            data = {**data, **add_kwargs}
+        add_kwargs = filter_missing(
+            {
+                "offset": offset,
+                "limit": limit,
+            }
+        )
+        data = {
+            **data,
+            **add_kwargs,
+        }
```

Comment:
I was under the impression that `data` would have `MISSING` filtered already, so this would be implicitly the same as:

```suggestion
        data = {
            "offset": offset,
            "limit": limit,
            **data,
        }
```

Note that I've swapped put `data` second, so that if a user puts `offset` or `limit` into their own payload, it will overwrite the default `MISSING` value.

If that's accurate, I think other places in the code can be updated/simplified in a similar way.

Reply:
You are correct, `MISSING` is filtered out at the time of sending the request. The reason I'm doing the above logic is to account for a case like this:
```
search_client.post_search(SearchQuery(), limit=100)
```
In the proposed solution above where `data` is constructed like so:
```
data = {
    "offset": offset,
    "limit": limit,
    **data,
}
```
`limit` will end up being `MISSING`, since `SearchQuery.limit` defaults to `MISSING`.

The old behavior of this method was that explicitly passed query params take precedence over query params set in `data`. The current implementation was to keep the same behavior by making an `add_kwargs` dict that does not have any `MISSING` values and use that to update `data`.

For reference, this was the old behavior of post search:
```
add_kwargs = {}
if offset is not None:
    add_kwargs["offset"] = offset
if limit is not None:
    add_kwargs["limit"] = limit
if add_kwargs:
    data = {**data, **add_kwargs}
```

Correct me if I'm wrong here, but I think the current proposed version is correct (consistent with the current SDK).

Reply:
Hm; okay, I need to stare at this a little to make a good recommendation.

You're definitely correct about the order of overrides in SDK v3 and what I would consider desirable behavior for `search_client.post_search(SearchQuery(), limit=100)`. I don't like the path this puts us on for supporting it though.

The issue, IMO, is that the ordering here is pretty delicate in order to get the effect of
- we don't modify the user input object
- we merge with a parameter if one is provided
- we don't merge if no parameter is provided

I'm trying to work out how the code here should look, and then we can "backfill" the implementation.
So, one example:
```python
# _copy_with is a method private to the SDK for doing
# copy+set on payload types with missing-awareness?
data = data._copy_with(limit=limit, offset=offset)
```
Or, very similar:
```python
# merge_parameters could have a signature which takes `*source_params: dict | PayloadWrapper`
# and then merges them in precedence order
data = utils.merge_parameters(data, {"limit": limit, "offset": offset})
```

I think this is subtle enough and high enough value that we should talk through a design for a solution.

Reply:
Per our discussion with the rest of the team, usages of `query_params` and `additional_fields` have been updated to now all take precedence over explicitly passed parameters.